qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -0,0 +1,591 @@
+        """
+        if qargs is None:
+            qubits = range(self.data.num_qubits)
+        else:
+            qubits = qargs
+
+        outcome = ['X'] * len(qubits)
+        outcome_prob = 1.0
+        probs = {}  # probabilities dictionary
+
+        self._get_probablities(qubits, outcome, outcome_prob, probs)
+
+        if decimals is not None:
+            for key, value in probs.items():
+                probs[key] = round(value, decimals)
+
+        return probs
+
+    def reset(self, qargs=None):
+        """Reset state or subsystems to the 0-state.
+
+        Args:
+            qargs (list or None): subsystems to reset, if None all
+                                  subsystems will be reset to their 0-state
+                                  (Default: None).
+
+        Returns:
+            StabilizerState: the reset state.
+
+        Additional Information:
+            If all subsystems are reset this will return the ground state
+            on all subsystems. If only some subsystems are reset this
+            function will perform a measurement on those subsystems and
+            evolve the subsystems so that the collapsed post-measurement
+            states are rotated to the 0-state. The RNG seed for this
+            sampling can be set using the :meth:`seed` method.
+        """
+        # Resetting all qubits does not require sampling or RNG
+        if qargs is None:
+            return StabilizerState(Clifford((np.eye(2 * self.data.num_qubits))))
+
+        for qubit in qargs:
+            # Apply measurement and get classical outcome
+            outcome = self._measure_and_update(qubit, 0)
+
+            # Use the outcome to apply X gate to any qubits left in the
+            # |1> state after measure, then discard outcome.
+            if outcome == 1:
+                _append_x(self.data, qubit)
+
+        return self
+
+    def measure(self, qargs=None):
+        """Measure subsystems and return outcome and post-measure state.
+
+        Note that this function uses the QuantumStates internal random
+        number generator for sampling the measurement outcome. The RNG
+        seed can be set using the :meth:`seed` method.
+
+        Args:
+            qargs (list or None): subsystems to sample measurements for,
+                                  if None sample measurement of all
+                                  subsystems (Default: None).
+
+        Returns:
+            tuple: the pair ``(outcome, state)`` where ``outcome`` is the
+                   measurement outcome string label, and ``state`` is the
+                   collapsed post-measurement stabilizer state for the
+                   corresponding outcome.
+        """
+        if qargs is None:
+            qargs = range(self.data.num_qubits)
+
+        outcome = ''
+        for qubit in qargs:
+            randbit = self._rng.randint(2)
+            outcome = str(self._measure_and_update(qubit, randbit)) + outcome
```

Comment:
Same here, measurement shouldn't have side effects on the current state. Though you will have to be a bit careful about rng when making copies so it works as expected if you set a seed but loop over calling measure on a state to sample outcomes.